src/ImageSharp/Image/ImageBase{TColor}.cs

```diff
@@ -162,13 +162,15 @@ public virtual PixelAccessor<TColor> Lock()
         internal void SwapPixelsBuffers(PixelAccessor<TColor> pixelSource)
         {
             Guard.NotNull(pixelSource, nameof(pixelSource));
-            Guard.IsTrue(pixelSource.PooledMemory, nameof(pixelSource.PooledMemory), "pixelSource must be using pooled memory");
 
+            // TODO: Was this check really useful? If yes, we can define a bool PixelAccessor<TColor>.IsBoundToImage to re-introduce it:
```

Comment:
The logic was that there was nothing stopping you passing a PixelAcessor that was created by calling  Lock on an array and you shouldn't be able to switch out the backing buffer from the accessor and the image then as one wouldn't have originated from the array pool thus it shouldn't later be returned to it.